pkg/envoy/eds/cluster_load_assignment.go

```diff
@@ -47,7 +50,27 @@ func newClusterLoadAssignment(svc service.MeshService, serviceEndpoints []endpoi
 				Value: weight,
 			},
 		}
-		cla.Endpoints[0].LbEndpoints = append(cla.Endpoints[0].LbEndpoints, &lbEpt)
+
+		// endpoint with Weight set is a remote cluster endpoint.
+		if meshEndpoint.Weight == 0 {
+			localLbEndpoints.LbEndpoints = append(localLbEndpoints.LbEndpoints, &lbEpt)
+		} else {
+			lbEpt.LoadBalancingWeight.Value = uint32(meshEndpoint.Weight)
+			remoteLbEndpoints := &xds_endpoint.LocalityLbEndpoints{
+				Locality: &xds_core.Locality{
+					Zone: meshEndpoint.Zone,
+				},
+				LbEndpoints: []*xds_endpoint.LbEndpoint{&lbEpt},
+				Priority:    remoteClusterPriority,
+				LoadBalancingWeight: &wrappers.UInt32Value{
+					Value: uint32(meshEndpoint.Weight),
+				},
+			}
+			if meshEndpoint.Priority != 0 {
+				remoteLbEndpoints.Priority = uint32(meshEndpoint.Priority)
+			}
+			cla.Endpoints = append(cla.Endpoints, remoteLbEndpoints)
```

Comment:
Seems like this can be simplified by changing the if-else clause as:
```
if meshEndpoint.Weight == 0 {
    // do something for local cluster endpoint
    continue
}

// do something for remote cluster endpoint
```